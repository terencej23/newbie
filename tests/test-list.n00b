
define function main with no params
	set a to [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
	print("Searching for element 12...")
	print(binary_search(a, 0, 9, 12))

	print("Searching for element 3...")
	print(binary_search(a, 0, 9, 3))



define function binary_search with params(a, low, high, key)
	if (high >= low)
		set mid to (low + high) / 2

		if (a[mid] = key)
			return mid
		if (a[mid] > key)
			return binary_search(a, low, mid-1, key)
		return binary_search(a, mid+1, high, key)

	return -1
