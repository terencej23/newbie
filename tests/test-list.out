Searching for element 12...
-1
Searching for element 3...
2